docs/topics/OAuth2.md

```diff
@@ -56,68 +41,229 @@ Scopes provide access to certain resources of a user's account. Your API client
+While Discord does not require the use of the `state` parameter, we support it and highly recommend that you implement it for the security of your own applications and data.
 
-### Two-Factor Authentication Requirement
+## Authorization Code Grant
 
-For bots with [elevated permissions](#DOCS_PERMISSIONS/bitwise-permission-flags) (permissions with a * next to them), we enforce two-factor authentication for the owner's account when used on guilds that have server-wide 2FA enabled.
+The authorization code grant is is what most developers will recognize as "standard OAuth2" and involves retrieving an access code and exchanging it for a user's access token. It allows the authorization server to act as an intermediary between the client and the resource owner, so the resource owner's credentials are never shared directly with the client.
 
-### Adding Bots to Guilds
+###### Authorization Code URL Example
 
-A URL can be generated that redirects authenticated users to the add-bot flow. It can be in the following format without a `redirect_uri`:
+```
+https://discordapp.com/oauth2/authorize?response_type=code&client_id=157730590492196864&scope=identify%20guilds.join&state=15773059ghq9183habn&redirect_uri=https%3A%2F%2Fnicememe.website
+```
 
+`client_id` is your application's `client_id`. `scope` is a list of [OAuth2 scopes](#DOCS_OAUTH2/shared-resources-oauth2-scopes) separated by url encoded spaces (`%20`). `redirect_uri` is whatever URL you registered when creating your application, url-encoded.
+
+When someone navigates to this URL, they will be prompted to authorize your application for the requested scopes. On acceptance, they will be redirected to your `redirect_uri`, which will contain an additional querystring parameter, `code`. This should be exchanged for the user's access token by making a `POST` request to the [token URL](#DOCS_OAUTH2/shared-resources-oauth2-urls) with the following parameters:
+
+- `client_id` - your application's client id
+- `client_secret` - your application's client secret
+- `grant_type` - must be set to `authorization_code`
+- `code` - the code from the querystring
+- `redirect_uri` - your `redirect_uri`
+
+###### Access Token Exchange Example
+
+```python
+def exchange_code(code, state):
+  data = {
+    'client_id': '332269999912132097',
+    'client_secret': '456265548123452097',
+    'grant_type': 'authorization_code',
+    'code': code,
+    'redirect_uri': 'https://nicememe.website',
+    'state': state
+  }
+  headers = {
+    'Content-Type': 'application/x-www-form-urlencoded'
+  }
+  r = requests.post('https://discordapp.com/api/oauth2/token', data, headers)
+  r.raise_for_status()
+  return r.json()
 ```
-https://discordapp.com/api/oauth2/authorize?client_id=157730590492196864&scope=bot&permissions=0
+
+You can also pass your `client_id` and `client_secret` as an authentication header of type `Basic` with `client_id` as the username and `client_secret` as the password. In response, you will receive:
```

Comment:
nit: `as basic authentication with` since otherwise the header comment makes no sense in the scope of user/pass